### PR TITLE
Include narinfo information in nixtract component output

### PR DESCRIPTION
**Pull request by ErinvanderVeen**
Resolves https://github.com/tweag/nixtract/issues/29

### CODE REVIEW

File: src/nix/substituters.rs
Comment (thufschmitt) [inline src/nix/substituters.rs:11]:
```suggestion
    let expr = "flakeRef: (import ((builtins.getFlake flakeRef).outPath + \"/flake.nix\")).nixConfig.extra-substituters or []";
```

and use `.args(["--argstr", flake_ref])` below

Comment (thufschmitt) [reply src/nix/substituters.rs:11]: Arrrgh yes, I thought about just that over the week-end and forgot to add that.

You can work around it with `nix eval --expr "{ flakeRef }: { result = blah; }.result" result` but that start s to be painfully involved :/

File: src/nix/narinfo.rs
Comment (thufschmitt) [inline src/nix/narinfo.rs:69]: That's probably too costly (and if not you'll hate me for suggesting it that late), but mentioning it any way for completeness: running `nix path-info --json --store {binaryCacheUrl}` can get you the store path information in json, and for arbitrary stores (including the local one, which can come in handy I guess)

File: src/nix/narinfo.rs
Comment (thufschmitt) [inline src/nix/narinfo.rs:150]: Probably don't want to fail here for forward-compat (and because Nix allows extra fields, so there might be in the wild)

File: src/nix/narinfo.rs
Comment (thufschmitt) [inline src/nix/narinfo.rs:164]: These also don't seem to be mandatory in Nix

File: src/nix/substituters.rs
Comment (ErinvanderVeen) [reply src/nix/substituters.rs:11]: I'll stick with the string interpolation right now, and try to switch to `nix-instantiate` after.

File: src/nix/narinfo.rs
Comment (thufschmitt) [inline src/nix/narinfo.rs:79]: If I'm not mistaken, the `Substituters` (at least as they are returned by `get_substituters`) are full store URLs. This means that:
- http(s) binary caches will already have the `https://` scheme included and shouldn't have it added
- Non-https ones won't work at all and should be dropped

Comment (thufschmitt) [reply src/nix/narinfo.rs:79]: Maybe this also hints at a missing test ;)

File: src/nix/narinfo.rs
Comment (ErinvanderVeen) [reply src/nix/narinfo.rs:174]: Is there a place where I can concretely see which fields are and aren't expected by Nix? I could ofc just assume all are optional, but that limited what kind of stuff we can do with them.

Comment (thufschmitt) [reply src/nix/narinfo.rs:174]: My main source was https://github.com/nixos/nix/blob/0d26358bda1637e54ebacac18e7b5af7381cf5f3/src/libstore/nar-info.cc#L98-L106

Comment (ErinvanderVeen) [reply src/nix/narinfo.rs:174]: Thank you <3

File: src/nix/narinfo.rs
Comment (ErinvanderVeen) [reply src/nix/narinfo.rs:69]: The current approach prevents me from having to go out of Rust and spawn another process. I will leave it as it is, until we encounter an issue with the current approach.

**Review by thufschmitt — APPROVED** [review]
:rocket:

**Comment by thufschmitt** [comment]
(I don't have merge rights, it seems. But you have my moral blessing to merge)

**Pull request merged**
